docs/developer/architecture/authentication.rst

```diff
@@ -0,0 +1,97 @@
+Authentication Architecture Document
+====================================
+
+Context:
+--------
+Currently, when we deploy MetalK8s we pre provision super admin user with admin
+password. Meaning anyone wanting to use K8S/Salt APIs should authenticate using
+this single super admin user. Another way to access the APIs is by using K8S
+admin certificate that is stored in /etc/kubernetes/admin.conf. We could also
+manually provisioned other users and credentials as well as roles binding
+but the way the functionality is accessible currently is too complex and not
+enough secured since user / password are stored in clear text.
+This is not good enough in term of security and flexibility.
+
+We would like at least to be able to add different users with different
+credentials and ideally to integrate K8S authentication system with external
+identity provider.
+
+Managing K8S role binding between user / groups High level roles and K8S roles
+is not part of this specification.
+
+Requirements:
+-------------
+
+Basically we are talking about:
+
+- Being able to provision users with an embedded Identity provider (IDP)
+- Being able to integrate with an external IDP
+
+`Dex <https://github.com/dexidp/dex>`_ seems to be the preferred solution for
+now, for both local identity provider or proxy to external IDP.
```

Comment:
Mentions of `Dex` in the requirements seems inappropriate. To me, `Dex` is _one_ solution to the set of requirements, and this choice should be documented in the "design" section of this architecture document, along with the discarded alternatives.

Reply:
ok so will replace DEX with local IDP